### DISCUSSION TITLE
Help needed for Username Password Authorizer

### DISCUSSION THREAD

**Discussion by daud-silvergrey** [Keyfactor/signserver-ce#55] [Q&A]
Hi,

I have set up a PDF Signer worker with the following configuration:

- Username Password Authorizer
- AUTHTYPE | org.signserver.server.UsernamePasswordAuthorizer

I am attempting to configure user accounts as follows:

- USER.[NAME] = [PASSWORD]
- USER.[NAME] = [HASHED_PASSWORD]:[HASH_ALGORITHM]
- USER.[NAME] = [HASHED_PASSWORD]:[HASH_ALGORITHM]:[SALT]

However, I am encountering difficulties in implementing this. Please assist me in configuring user accounts using these properties.

**Comment by netmackan** [comment]
Example from the unit tests:
https://github.com/Keyfactor/signserver-ce/blob/6f53e0ddccccd360b257ef0f397105a45e3d180c/signserver/modules/SignServer-Test-System/src/test/java/org/signserver/server/UsernamePasswordAuthorizerTest.java#L80-L83

In this case:
Username: user3
Password: foo123
Salt: salt123
USER.USER3=26c110963ad873c9b7db331e4c3130c266416d47:SHA1:salt123

Where you can for instance compute the value using:
```
$ echo -n "foo123salt123" | sha1sum
```

Note: a stronger hash algorithm should be used or if higher security or flexibility is wanted to instead consider using a reverse proxy in front and do the authentication there.
